src/components/legend/draw.js

```diff
@@ -568,18 +568,32 @@ function computeLegendDimensions(gd, groups, traces) {
     else {
         opts.width = 0;
         opts.height = 0;
+        var rowHeight = 0;
+        var maxTraceHeight = 0;
+        var startX = 0;
```

Comment:
Also, I rename `startX` to `offsetX` to be consistent with the naming used [here](https://github.com/psalmody/plotly.js/blob/5b5a102874974be5d726bc7ef6edf54c16746de7/src/components/legend/draw.js#L520)